

<launch>
<!-- top_sync -->

<!-- set -->
<param name="pitch_calibration" value="true" />

<!-- remove -->
<param name="AREA_THRES" value="0.2" />
<param name="AREA_GAIN" value="1" />
<param name="CAR_HEIGHT_" value="-1.68" />
<param name="ALPHA_MAX_" value="20" />
<param name="H_MIN_" value="0.15" />
<param name="init_ALPHA_MAX_" value="6" />
<param name="Long_Distance" value="45" />
<param name="SetUnderZ" value="1" />
<param name="SET_GROUND_DEG" value ="10" />

<!-- each -->
<param name="XYTHRESHOLD" value="0.3" />
<param name="CenterThreshold" value="1.5" />
<param name="LRThreshold" value="1.0" />

<!-- segment -->
<param name="THRESHOLD" value="0.5" />
<param name="MIN_OBJECT_NUM" value="1" />
<param name="AfterSetHeight" value="1" />

<!-- LRTcombine -->
<param name="LR_THRSHOLD" value="0.5" />
<param name="CENTER_THRSHOLD" value="0.5" />
<param name="LimitCenter" value="10" />
<param name="pose_gap" value="0.015" />

<!-- pharos_msg -->
<param name="SET_UNDER_Z" value="1.0" />
<param name="remove_point_size" value="5" />
<param name="Set_bong_distance" value="1" />
<param name="Bigvolum" value="300" />
<param name="map_size" value="100" />
<param name="Set_Perception_Distance" value="12" />
<param name="XY_AREA" value="0.45" />

<!-- set_topic_select -->
<arg name="topic_name" default="my" />


<node pkg="pharos_vlp_tilt"         type="vlpt_set_T"     name="vlpt_set_T" >

		<param name="T_PLUS_X" value="2.5" />
		<param name="T_MINUS_X" value="-1.8" />

		<param name="T_PLUS_Y" value="1" />
		<param name="T_MINUS_Y" value="-1" />	

		<param name="T_PLUS_Z" value="20" />
		<param name="T_MINUS_Z" value="-20" />	
		<param name="T_MOVE_Y" value="0.5265" />

		<param name="sub_topic" value="vlp_t/my_custom_point_info" />
		<param name="pub_topic" value="top_my_msg" />
		<param name="frame_my" value="top_velodyne2" />
		<param name="CAR_HEIGHT" value="-1.68" />


	</node>


	<node pkg="pharos_vlp_tilt"         type="vlpt_remove_ground_T"     name="vlpt_remove_ground_T" >
		<param name="sub_topic" value="top_$(arg topic_name)_msg" />
		<param name="pub_topic_ground" value="ground_t" />
		<param name="pub_topic_objects" value="object_t" />
		<param name="pub_topic_my_msg" value="before_t_object" />
		<param name="frame_remove" value="top_velodyne2" />	
		
	
	</node>

	<node pkg="pharos_vlp_tilt"         type="vlpt_filter_T"     name="vlpt_filter_T" >
		<param name="map_topic" value="drivable_map_perception_map" />
		<param name="odom_topic" value="/odom/vehicle" />
		
	</node>

	<node pkg="pharos_vlp_tilt"         type="vlpt_segmentation_T"     name="vlpt_segmentation_T">
		<param name="frame_id" value="top_velodyne2" />
		<param name="sub_topic_name_" value="t_into_segmentation" />	
		<param name="pub_topic_name_" value="my_t_seg" />
		
	</node>

	<node pkg="pharos_vlp_tilt"         type="vlpt_each_T"     name="vlpt_each_T">
		<param name="frame_id_" value="top_velodyne2" />
		<param name="pub_topic_name_" value="new_top_combine" />	
		<param name="sub_topic_name_" value="my_t_seg" />
		
	</node>


	
	<node pkg="pharos_vlp_tilt"         type="vlpt_set_L"     name="vlpt_set_L" >

		<param name="L_PLUS_X" value="2.5" />
		<param name="L_MINUS_X" value="-1.8" />

		<param name="L_PLUS_Y" value="0.4" />
		<param name="L_MINUS_Y" value="-1.5" />	

		<param name="L_PLUS_Z" value="20" />
		<param name="L_MINUS_Z" value="-20" />	
		<param name="L_MOVE_Y" value="0.5265" />

		<param name="sub_topic" value="vlp_l/my_custom_point_info" />
		<param name="pub_topic" value="left_my_msg" />
		<param name="frame_my" value="left_velodyne2" />
		<param name="CAR_HEIGHT" value="-1.62" />


	</node>


	<node pkg="pharos_vlp_tilt"         type="vlpt_remove_ground_L"     name="vlpt_remove_ground_L" >
		<param name="sub_topic" value="left_$(arg topic_name)_msg" />
		<param name="pub_topic_ground" value="ground_l" />
		<param name="pub_topic_objects" value="object_l" />
		<param name="pub_topic_my_msg" value="before_l_object" />
		<param name="frame_remove" value="left_velodyne2" />	
		
		
	
	</node>

	<node pkg="pharos_vlp_tilt"         type="vlpt_filter_L"     name="vlpt_filter_L" >
		<param name="map_topic" value="drivable_map_perception_map" />
		<param name="odom_topic" value="/odom/vehicle" />
		
	</node>

	<node pkg="pharos_vlp_tilt"         type="vlpt_segmentation_L"     name="vlpt_segmentation_L" >
		<param name="frame_id" value="left_velodyne2" />
		<param name="sub_topic_name_" value="l_into_segmentation" />	
		<param name="pub_topic_name_" value="my_l_seg" />
		
	</node>

	<node pkg="pharos_vlp_tilt"         type="vlpt_each_L"     name="vlpt_each_L" >
		<param name="frame_id_" value="left_velodyne2" />
		<param name="pub_topic_name_" value="new_left_combine" />	
		<param name="sub_topic_name_" value="my_l_seg" />
		
	</node>




	
	<node pkg="pharos_vlp_tilt"         type="vlpt_set_R"     name="vlpt_set_R" >

		<param name="R_PLUS_X" value="2.5" />
		<param name="R_MINUS_X" value="-1.8" />

		<param name="R_PLUS_Y" value="1.2" />
		<param name="R_MINUS_Y" value="-0.2" />	

		<param name="R_PLUS_Z" value="20" />

		<param name="R_MINUS_Z" value="-20" />	
	
		<param name="R_MOVE_Y" value="-0.5265" />

		<param name="sub_topic" value="vlp_r/my_custom_point_info" />
		<param name="pub_topic" value="right_my_msg" />
		<param name="frame_my" value="right_velodyne2" />
		<param name="CAR_HEIGHT" value="-1.62" />

	</node>

	<node pkg="pharos_vlp_tilt"         type="vlpt_remove_ground_R"     name="vlpt_remove_ground_R"    >
		<param name="sub_topic" value="right_$(arg topic_name)_msg" />
		<param name="pub_topic_ground" value="ground_r" />
		<param name="pub_topic_objects" value="object_r" />
		<param name="pub_topic_my_msg" value="before_r_object" />
		<param name="frame_remove" value="right_velodyne2" />	
		
	</node>

	<node pkg="pharos_vlp_tilt"         type="vlpt_filter_R"     name="vlpt_filter_R" >
		<param name="map_topic" value="drivable_map_perception_map" />
		<param name="odom_topic" value="/odom/vehicle" />
	</node>


	<node pkg="pharos_vlp_tilt"         type="vlpt_segmentation_R"     name="vlpt_segmentation_R">
		<param name="frame_id" value="right_velodyne2" />
		<param name="sub_topic_name_" value="r_into_segmentation" />	
		<param name="pub_topic_name_" value="my_r_seg" />
		
	</node>


<node pkg="pharos_vlp_tilt"         type="vlpt_each_R"     name="vlpt_each_R" >
		<param name="frame_id_" value="right_velodyne2" />
		<param name="pub_topic_name_" value="new_right_combine" />	
		<param name="sub_topic_name_" value="my_r_seg" />
		
	</node>




<node pkg="pharos_vlp_tilt"         type="vlpt_LRT_Combine"     name="vlpt_LRT_Combine" >

		<param name="all_frame" value="vehicle_frame" />
		

	</node>

<node pkg="pharos_vlp_tilt"         type="vlpt_pharos_msg"     name="vlpt_pharos_msg" >

			<param name="all_frame" value="odom" />
			<param name="odom_topic_name" value="/odom/vehicle" />
			<param name="map_topic_name" value= "drivable_map_perception_map"/>
    </node>


</launch>
